R/hazardFunctions.R

```diff
@@ -19,3 +19,40 @@ ExpHazOS <- function(t, h01, h02, h12) {
   h012 <- h12 - h01 - h02
   ((h12 - h02) * (h01 + h02) - h01 * h12 * exp(-h012 * t)) / ((h12 - h02) - h01 * exp(-h012 * t))
 }
+
+#' Average OS Hazard Ratio from Constant Transition Hazards
+#'
+#' @param transitionByArm (`list`) \cr transition parameters for each treatment group.
+#'   See [exponential_transition()], [piecewise_exponential()] and [weibull_transition()] for details.
+#' @param alpha assigns weights to time points, where values higher than 0.5 assign greater weight
+#' to earlier times and values lower than 0.5 assign greater weight to later times.
+#' @param upper upper (time) limit of integration
+#'
+#' @return This returns the value of the average hazard ratio
+#' @export
+#'
+#' @examples
+#' transitionTrt <- exponential_transition(h01 = 0.18, h02 = 0.06, h12 = 0.17)
+#' transitionCtl <- exponential_transition(h01 = 0.23, h02 = 0.07, h12 = 0.19)
+#' transitionList <- list(transitionCtl, transitionTrt)
+#' ahrExpOS(transitionByArm = transitionList, alpha = 0.5, upper = 100)
+ahrExpOS <- function(transitionByArm, alpha = 0.5, upper = Inf) {
```

Comment:
```suggestion
avgHRExpOS <- function(transitionByArm, alpha = 0.5, upper = Inf) {
```
could be easier to read name? or something similar?